apps/examples/wifi_manager_sample/wm_test_stress_test3.c

```diff
@@ -0,0 +1,280 @@
+static void wm_scan_done(wifi_manager_scan_info_s **scan_result, wifi_manager_scan_result_e res);
+
+static wifi_manager_cb_s g_wifi_callbacks = {
+	wm_sta_connected,
+	wm_sta_disconnected,
+	wm_softap_sta_join,
+	wm_softap_sta_leave,
+	wm_scan_done,
+};
+
+#define WM_TEST_SIGNAL							\
+	do {										\
+		sem_post(&g_wm_sem);					\
+		WT_LOG(WT_TAG, "send signal");			\
+	} while (0)
+
+#define WM_TEST_WAIT							\
+	do {										\
+		WT_LOG(WT_TAG, "wait signal");			\
+		sem_wait(&g_wm_sem);					\
+	} while (0)
+
+void wm_sta_connected(wifi_manager_result_e res)
+{
+	WT_LOG(WT_TAG, "--> res(%d)", res);
+	WM_TEST_SIGNAL;
+}
+
+void wm_sta_disconnected(wifi_manager_disconnect_e disconn)
+{
+	WT_LOG(WT_TAG, "-->");
+	WM_TEST_SIGNAL;
+}
+
+void wm_softap_sta_join(void)
+{
+	WT_LOG(WT_TAG, "-->");
+	WM_TEST_SIGNAL;
+}
+
+void wm_softap_sta_leave(void)
+{
+	WT_LOG(WT_TAG, "-->");
+	WM_TEST_SIGNAL;
+}
+
+void wm_scan_done(wifi_manager_scan_info_s **scan_result, wifi_manager_scan_result_e res)
+{
+	WT_LOG(WT_TAG, "-->");
+	/* Make sure you copy the scan results onto a local data structure.
+	 * Int will be deleted soon eventually as you exit this function.
+	 */
+	if (scan_result == NULL) {
+		WM_TEST_SIGNAL;
+		return;
+	}
+	wifi_manager_scan_info_s *wifi_scan_iter = *scan_result;
+	while (wifi_scan_iter != NULL) {
+		WT_LOG(WT_TAG, "WiFi AP SSID: %-20s, WiFi AP BSSID: %-20s, WiFi Rssi: %d, AUTH: %d, CRYPTO: %d",
+			   wifi_scan_iter->ssid, wifi_scan_iter->bssid, wifi_scan_iter->rssi,
+			   wifi_scan_iter->ap_auth_type, wifi_scan_iter->ap_crypto_type);
+		wifi_scan_iter = wifi_scan_iter->next;
+	}
+	WM_TEST_SIGNAL;
+}
+
+static void wm_get_softapinfo(wifi_manager_softap_config_s *ap_config)
+{
+	strncpy(ap_config->ssid, WM_SOFTAP_SSID, strlen(WM_SOFTAP_SSID) + 1);
+	strncpy(ap_config->passphrase, WM_SOFTAP_PASSWORD, strlen(WM_SOFTAP_PASSWORD) + 1);
+	ap_config->channel = WM_SOFTAP_CHANNEL;
+}
+
+static void wm_get_apinfo(wifi_manager_ap_config_s *apconfig)
+{
+	strncpy(apconfig->ssid, WM_AP_SSID, strlen(WM_AP_SSID) + 1);
+	apconfig->ssid_length = strlen(WM_AP_SSID);
+	apconfig->ap_auth_type = WM_AP_AUTH;
+	if (WM_AP_AUTH != WIFI_MANAGER_AUTH_OPEN) {
+		strncpy(apconfig->passphrase, WM_AP_PASSWORD, strlen(WM_AP_PASSWORD) + 1);
+		apconfig->passphrase_length = strlen(WM_AP_PASSWORD);
+		apconfig->ap_crypto_type = WM_AP_CRYPTO;
+	}
+}
+
+static int _run_procedure(void)
+{
+	int nres = 0;
+	wifi_manager_result_e wres = WIFI_MANAGER_SUCCESS;
+	WT_LOG(WT_TAG, "init wi-fi");
+	wres = wifi_manager_init(&g_wifi_callbacks);
+	if (wres != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(WT_TAG, "fail to init %d\n", wres);
+		return -1;
+	}
+
+	/*  Start softAP */
+	WT_LOG(WT_TAG, "start softAP");
+	wifi_manager_softap_config_s softap_config;
+	wm_get_softapinfo(&softap_config);
+	wres = wifi_manager_set_mode(SOFTAP_MODE, &softap_config);
+	if (wres != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(WT_TAG, "fail to start softap %d\n", wres);
+		return -1;
```

Comment:
Is it okay not to deinit the `wifi manager`?

Reply:
As you commented, it would be more safe to call wifi_deinit here. However it's test code. I think return -1 is enough to test.